test/classification/test_snorkel_classifier.py

```diff
@@ -2,38 +2,83 @@
 import tempfile
 import unittest
 
+import numpy as np
 import torch
 import torch.nn as nn
 
+from snorkel.classification.data import DictDataLoader, DictDataset
 from snorkel.classification.scorer import Scorer
 from snorkel.classification.snorkel_classifier import Operation, SnorkelClassifier, Task
 
+NUM_EXAMPLES = 10
+BATCH_SIZE = 2
 
-class TaskTest(unittest.TestCase):
+
+def create_dataloader(task_name="task", split="train", **kwargs):
+    X = torch.FloatTensor([[i, i] for i in range(NUM_EXAMPLES)])
+    Y = torch.ones(NUM_EXAMPLES, 1).long()
+
+    dataset = DictDataset(
+        name="dataset", split=split, X_dict={"data": X}, Y_dict={task_name: Y}
+    )
+
+    dataloader = DictDataLoader(dataset, batch_size=BATCH_SIZE, **kwargs)
+    return dataloader
+
+
+def create_task(task_name, module_suffixes=("", "")):
+    module1_name = f"linear1{module_suffixes[0]}"
+    module2_name = f"linear2{module_suffixes[1]}"
+
+    module_pool = nn.ModuleDict(
+        {
+            module1_name: nn.Sequential(nn.Linear(2, 4), nn.ReLU()),
```

Comment:
Is it worth it to manually set the weights so we know expected outputs of a forward pass? maybe just to the identity?

Reply:
Hmm, yeah. I'll give it a try.